src/fromager/dependencies.py

```diff
@@ -110,7 +110,7 @@ def default_get_build_backend_dependencies(
     ctx: context.WorkContext,
     req: Requirement,
     sdist_root_dir: pathlib.Path,
-) -> set[Requirement]:
+) -> typing.Iterable[str]:
```

Comment:
Are you sure this is correct? `_filter_requirements` returns a set of Requirements.

Reply:
The return statement of this function is one of the hooks method: https://github.com/python-wheel-build/fromager/pull/234/files#diff-37a100b3c15c4653f7f126670e7e58c41fc41074761085dd21e375b905125b51L119

And the docs says that it returns a Sequence so I should probably change it to that: https://pyproject-hooks.readthedocs.io/en/latest/pyproject_hooks.html#pyproject_hooks.BuildBackendHookCaller.get_requires_for_build_editable